src/ast/ddl.rs

```diff
@@ -330,6 +352,64 @@ impl fmt::Display for TableConstraint {
 
                 Ok(())
             }
+            Self::FulltextOrSpatial {
+                fulltext,
+                index_type_display,
+                opt_index_name,
+                columns,
+            } => {
+                if *fulltext {
+                    write!(f, "FULLTEXT")?;
+                } else {
+                    write!(f, "SPATIAL")?;
+                }
+
+                if !matches!(index_type_display, KeyOrIndexDisplay::None) {
+                    write!(f, " {}", index_type_display)?;
+                }
```

Comment:
Since `KeyOrIndexDisplay::None` is displayed as `""` I suspect you don't even have to have this test

```suggestion
                write!(f, " {}", index_type_display)?;
```

Reply:
@alamb the problem is that i'd have an additional space compared to the input string. Although it isn't incorrect, IMHO would be weird to print something like:

```sql
CREATE TABLE tb (c1 TEXT, FULLTEXT  (c1, c2);
```